stacker/util.py

```diff
@@ -554,24 +631,108 @@ def __init__(self, sources, stacker_cache_dir=None):
         self.create_cache_directories()
 
     def create_cache_directories(self):
-        """Ensures that SourceProcessor cache directories exist."""
+        """Ensure that SourceProcessor cache directories exist."""
         if not os.path.isdir(self.package_cache_dir):
             if not os.path.isdir(self.stacker_cache_dir):
                 os.mkdir(self.stacker_cache_dir)
             os.mkdir(self.package_cache_dir)
 
     def get_package_sources(self):
-        """Makes remote python packages available for local use."""
+        """Make remote python packages available for local use."""
+        # Checkout S3 repositories specified in config
+        for config in self.sources.get('s3', []):
+            self.fetch_s3_package(config=config)
         # Checkout git repositories specified in config
         for config in self.sources.get('git', []):
             self.fetch_git_package(config=config)
 
+    def fetch_s3_package(self, config):
+        """Make a remote S3 archive available for local use.
+
+        Args:
+            config (dict): git config dictionary
+
+        """
+        extractor_map = {'.tar.gz': TarGzipExtractor,
+                         '.tar': TarExtractor,
+                         '.zip': ZipExtractor}
+        extractor = None
+        for suffix, klass in extractor_map.iteritems():
+            if config['key'].endswith(suffix):
+                extractor = klass()
+                dir_name = self.sanitize_uri_path(
```

Comment:
Could you throw a debug statement here detailing the extractor chosen for the given path?